packages/snaps-execution-environments/src/common/endowments/network.ts

```diff
@@ -7,7 +7,7 @@ import type { EndowmentFactoryOptions } from './commonEndowmentFactory';
  * This class wraps a Response object.
  * That way, a teardown process can stop any processes left.
  */
-class ResponseWrapper implements Response {
+export class ResponseWrapper extends Response {
```

Comment:
Devs can still access the underlying `Response` by using the prototype chain. We'll need to fix that before merging.

Reply:
One trick that might allow us more control over what instanceof says is giving it a custom implementation

```js
class MyClass {
  static [Symbol.hasInstance](instance) {
    return (anything we want here)
  }
}
```

Reply:
That seems like a better solution that extending, instead we can replace the global `Response` class with:
```
class AlteredResponse extends Response {
  static [Symbol.hasInstance](instance: any) {
    return instance instanceof Response || instance instanceof ResponseWrapper;
  }
}
```